### ISSUE TITLE
EKS E2E tests job is broken

### ISSUE THREAD

**Issue by thbkrkr** [elastic/cloud-on-k8s#3196]
Since May 18, the job `cloud-on-k8s-e2e-tests-eks/` fails when creating the k8s cluster.

```
02:28:54  2020/05/19 00:28:45 Creating cluster ...
03:06:18  Error: timed out (after 25m0s) waiting for at least 3 nodes to join the cluster and become ready in "ng-1"
03:06:18  Error: exit status 1
```

**Comment by pebrc** [comment]
We need a Slack notification for this job

**Comment by thbkrkr** [comment]
The output is purposely hidden to avoid any leakage of sensitive data, so we are quite blind here.

I executed the cluster creation with a more detailed log level (`eksctl create cluster -v 4 -f eks.yml`):

```sh
...
2020-06-05T15:17:14+02:00 [▶]  start waiting for CloudFormation stack "eksctl-test-eks-thb-cluster"
2020-06-05T15:17:14+02:00 [▶]  waiting for CloudFormation stack "eksctl-test-eks-thb-cluster"
2020-06-05T15:17:31+02:00 [▶]  waiting for CloudFormation stack "eksctl-test-eks-thb-cluster"
2020-06-05T15:17:51+02:00 [▶]  waiting for CloudFormation stack "eksctl-test-eks-thb-cluster"
2020-06-05T15:18:10+02:00 [▶]  waiting for CloudFormation stack "eksctl-test-eks-thb-cluster"
2020-06-05T15:18:11+02:00 [✖]  unexpected status "ROLLBACK_IN_PROGRESS" while waiting for CloudFormation stack "eksctl-test-eks-thb-cluster"
...
2020-06-05T15:18:11+02:00 [✖]  AWS::EC2::SubnetRouteTableAssociation/RouteTableAssociationPublicEUWEST2A: CREATE_FAILED – "Resource creation cancelled"
2020-06-05T15:18:11+02:00 [✖]  AWS::EC2::NatGateway/NATGateway: CREATE_FAILED – "Resource creation cancelled"
2020-06-05T15:18:11+02:00 [✖]  AWS::EC2::SubnetRouteTableAssociation/RouteTableAssociationPrivateEUWEST2A: CREATE_FAILED – "Resource creation cancelled"
2020-06-05T15:18:11+02:00 [✖]  AWS::EC2::SubnetRouteTableAssociation/RouteTableAssociationPublicEUWEST2C: CREATE_FAILED – "Resource creation cancelled"
2020-06-05T15:18:11+02:00 [✖]  AWS::EC2::SubnetRouteTableAssociation/RouteTableAssociationPrivateEUWEST2C: CREATE_FAILED – "Resource creation cancelled"
2020-06-05T15:18:11+02:00 [✖]  AWS::EC2::SubnetRouteTableAssociation/RouteTableAssociationPrivateEUWEST2B: CREATE_FAILED – "Resource creation cancelled"
2020-06-05T15:18:11+02:00 [✖]  AWS::EC2::SubnetRouteTableAssociation/RouteTableAssociationPublicEUWEST2B: CREATE_FAILED – "Resource creation cancelled"
2020-06-05T15:18:11+02:00 [✖]  AWS::EKS::Cluster/ControlPlane: CREATE_FAILED – "The provided roleArn is invalid. (Service: AmazonEKS; Status Code: 400; Error Code: Invalid
ParameterException; Request ID: ab138db4-dba2-4f3d-9743-ae8b7f0f5449)"
...
```

**Comment by thbkrkr** [comment]
> We need a Slack notification for this job

I think we were waiting for the job to be "really stable" and it has been for a while, so yes!

**Comment by thbkrkr** [comment]
I think I had misconfigured the aws creds during my previous attempt to create a cluster. I tried again and this time it worked (in region eu-west-2).
I tried again using `ci-interactive` to be under the same conditions as the CI and then it fails:

```sh
2020-06-18T07:43:45Z [✖]  AWS::EC2::EIP/NATIP: CREATE_FAILED – 
  "Resource creation cancelled"
2020-06-18T07:43:45Z [✖]  AWS::EC2::VPC/VPC: CREATE_FAILED – 
  "The maximum number of VPCs has been reached. 
    (Service: AmazonEC2; Status Code: 400; Error Code: VpcLimitExceeded)"
2020-06-18T07:43:45Z [✖]  AWS::EC2::InternetGateway/InternetGateway: CREATE_FAILED – 
  "The maximum number of internet gateways has been reached. 
    (Service: AmazonEC2; Status Code: 400; Error Code: InternetGatewayLimitExceeded)"
```
We have a quota issue. The CI uses region `eu-west-1` where there are already 2 clusters. The default VPC limit is 5 VPCs per region. One cluster uses 2 VPC. We cannot create a third cluster.

**Comment by pebrc** [comment]
I created an additional cluster to test. I did not see the quota error when I did so but still the worker nodes from the node group where not joining the k8s cluster. I have not had the time to look any further. Happy to hand this over to you @thbkrkr if you want to take another look.

**Comment by thbkrkr** [comment]
Now that you said it, it seems pretty evident that the cluster creation hangs on waiting for nodes to become ready. The message we got is pretty clear about that in CI since the beginning:
```
11:46:34  Error: timed out (after 25m0s) waiting for at least 
          3 nodes to join the cluster and become ready in "ng-1"
```
But that gives us only a starting point. For now I can just say that we are not alone: https://github.com/weaveworks/eksctl/issues/1482.

**Comment by thbkrkr** [comment]
I don't know if it's a coincidence but this issue started on May 19th just after we updated eksctl from `0.16.0` to `0.19.0` in #3092.

**Comment by thbkrkr** [comment]
I created a new cluster with [ssh allowed](https://eksctl.io/usage/creating-and-managing-clusters/#using-config-files) and got the public IP with:
```sh
> aws ec2 describe-instances --region $region | \
  jq '.Reservations[] | .Instances[] | {kn:.KeyName,ip:.PublicIpAddress,private_dns:.PrivateDnsName}'
```
This gives me access to the kubelet log:

```
[ec2-user@ip-192-168-16-30 log]$ sudo tail -50 messages 
Jun 24 12:53:11 ip-192-168-16-30 kubelet: describe-cluster                         | describe-update
Jun 24 12:53:11 ip-192-168-16-30 kubelet: list-clusters                            | list-updates
Jun 24 12:53:11 ip-192-168-16-30 kubelet: update-cluster-version                   | update-kubeconfig
Jun 24 12:53:11 ip-192-168-16-30 kubelet: wait                                     | help
Jun 24 12:53:11 ip-192-168-16-30 kubelet: E0624 12:53:11.464107    3853 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.RuntimeClass: Get https://A5782ABEAD64D50D1C46F116D442503D.gr7.eu-central-1.eks.amazonaws.com/apis/node.k8s.io/v1beta1/runtimeclasses?limit=500&resourceVersion=0: getting credentials: exec: exit status 2
Jun 24 12:53:11 ip-192-168-16-30 kubelet: E0624 12:53:11.523852    3853 kubelet.go:2251] node "ip-192-168-16-30.eu-central-1.compute.internal" not found
Jun 24 12:53:11 ip-192-168-16-30 kubelet: E0624 12:53:11.623964    3853 kubelet.go:2251] node "ip-192-168-16-30.eu-central-1.compute.internal" not found
Jun 24 12:53:11 ip-192-168-16-30 kubelet: usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
Jun 24 12:53:11 ip-192-168-16-30 kubelet: To see help text, you can run:
Jun 24 12:53:11 ip-192-168-16-30 kubelet: aws help
Jun 24 12:53:11 ip-192-168-16-30 kubelet: aws <command> help
Jun 24 12:53:11 ip-192-168-16-30 kubelet: aws <command> <subcommand> help
Jun 24 12:53:11 ip-192-168-16-30 kubelet: aws: error: argument operation: Invalid choice, valid choices are:
Jun 24 12:53:11 ip-192-168-16-30 kubelet: create-cluster                           | delete-cluster
Jun 24 12:53:11 ip-192-168-16-30 kubelet: describe-cluster                         | describe-update
Jun 24 12:53:11 ip-192-168-16-30 kubelet: list-clusters                            | list-updates
Jun 24 12:53:11 ip-192-168-16-30 kubelet: update-cluster-version                   | update-kubeconfig
Jun 24 12:53:11 ip-192-168-16-30 kubelet: wait                                     | help
Jun 24 12:53:11 ip-192-168-16-30 kubelet: E0624 12:53:11.709523    3853 reflector.go:126] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://A5782ABEAD64D50D1C46F116D442503D.gr7.eu-central-1.eks.amazonaws.com/api/v1/pods?fieldSelector=spec.nodeName%3Dip-192-168-16-30.eu-central-1.compute.internal&limit=500&resourceVersion=0: getting credentials: exec: exit status 2
Jun 24 12:53:11 ip-192-168-16-30 kubelet: E0624 12:53:11.724109    3853 kubelet.go:2251] node "ip-192-168-16-30.eu-central-1.compute.internal" not found
Jun 24 12:53:11 ip-192-168-16-30 kubelet: E0624 12:53:11.824231    3853 kubelet.go:2251] node "ip-192-168-16-30.eu-central-1.compute.internal" not found
Jun 24 12:53:11 ip-192-168-16-30 kubelet: E0624 12:53:11.924370    3853 kubelet.go:2251] node "ip-192-168-16-30.eu-central-1.compute.internal" not found
Jun 24 12:53:11 ip-192-168-16-30 kubelet: usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
Jun 24 12:53:11 ip-192-168-16-30 kubelet: To see help text, you can run:
Jun 24 12:53:11 ip-192-168-16-30 kubelet: aws help
Jun 24 12:53:11 ip-192-168-16-30 kubelet: aws <command> help
Jun 24 12:53:11 ip-192-168-16-30 kubelet: aws <command> <subcommand> help
Jun 24 12:53:11 ip-192-168-16-30 kubelet: aws: error: argument operation: Invalid choice, valid choices are:
Jun 24 12:53:11 ip-192-168-16-30 kubelet: create-cluster                           | delete-cluster
Jun 24 12:53:11 ip-192-168-16-30 kubelet: describe-cluster                         | describe-update
Jun 24 12:53:11 ip-192-168-16-30 kubelet: list-clusters                            | list-updates
Jun 24 12:53:11 ip-192-168-16-30 kubelet: update-cluster-version                   | update-kubeconfig
Jun 24 12:53:11 ip-192-168-16-30 kubelet: wait                                     | help
Jun 24 12:53:11 ip-192-168-16-30 kubelet: E0624 12:53:11.952917    3853 reflector.go:126] k8s.io/kubernetes/pkg/kubelet/kubelet.go:442: Failed to list *v1.Service: Get https://A5782ABEAD64D50D1C46F116D442503D.gr7.eu-central-1.eks.amazonaws.com/api/v1/services?limit=500&resourceVersion=0: getting credentials: exec: exit status 2
Jun 24 12:53:12 ip-192-168-16-30 kubelet: E0624 12:53:12.024493    3853 kubelet.go:2251] node "ip-192-168-16-30.eu-central-1.compute.internal" not found
Jun 24 12:53:12 ip-192-168-16-30 kubelet: E0624 12:53:12.124615    3853 kubelet.go:2251] node "ip-192-168-16-30.eu-central-1.compute.internal" not found
Jun 24 12:53:12 ip-192-168-16-30 kubelet: usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
Jun 24 12:53:12 ip-192-168-16-30 kubelet: To see help text, you can run:
Jun 24 12:53:12 ip-192-168-16-30 kubelet: aws help
Jun 24 12:53:12 ip-192-168-16-30 kubelet: aws <command> help
Jun 24 12:53:12 ip-192-168-16-30 kubelet: aws <command> <subcommand> help
Jun 24 12:53:12 ip-192-168-16-30 kubelet: aws: error: argument operation: Invalid choice, valid choices are:
Jun 24 12:53:12 ip-192-168-16-30 kubelet: create-cluster                           | delete-cluster
Jun 24 12:53:12 ip-192-168-16-30 kubelet: describe-cluster                         | describe-update
Jun 24 12:53:12 ip-192-168-16-30 kubelet: list-clusters                            | list-updates
Jun 24 12:53:12 ip-192-168-16-30 kubelet: update-cluster-version                   | update-kubeconfig
Jun 24 12:53:12 ip-192-168-16-30 kubelet: wait                                     | help
Jun 24 12:53:12 ip-192-168-16-30 kubelet: E0624 12:53:12.197405    3853 event.go:200] Unable to write event: 'Patch https://A5782ABEAD64D50D1C46F116D442503D.gr7.eu-central-1.eks.amazonaws.com/api/v1/namespaces/default/events/ip-192-168-16-30.eu-central-1.compute.internal.161b7ac78d8b4f3b: getting credentials: exec: exit status 2' (may retry after sleeping)
Jun 24 12:53:12 ip-192-168-16-30 kubelet: E0624 12:53:12.224757    3853 kubelet.go:2251] node "ip-192-168-16-30.eu-central-1.compute.internal" not found
Jun 24 12:53:12 ip-192-168-16-30 kubelet: E0624 12:53:12.324881    3853 kubelet.go:2251] node "ip-192-168-16-30.eu-central-1.compute.internal" not found
```

It looks like there is an issue to get credentials using the aws cli.

```
[ec2-user@ip-192-168-16-30 log]$ aws --version
aws-cli/1.16.102 Python/2.7.16 Linux/4.14.154-128.181.amzn2.x86_64 botocore/1.12.92

[ec2-user@ip-192-168-16-30 log]$ aws eks get-token --cluster-name thb-debug-1- --region eu-central-1
usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
To see help text, you can run:

  aws help
  aws <command> help
  aws <command> <subcommand> help
aws: error: argument operation: Invalid choice, valid choices are:

create-cluster                           | delete-cluster                          
describe-cluster                         | describe-update                         
list-clusters                            | list-updates                            
update-cluster-version                   | update-kubeconfig                       
wait                                     | help  

thb@fun > aws --version                                                                                  
aws-cli/1.18.33 Python/3.6.9 Linux/4.15.0-96-generic botocore/1.15.33
                                                                                                                                                                        
thb@fun > aws eks get-token --cluster-name thb-debug-1- --region eu-central-1
{"kind": "ExecCredential", "apiVersion": "client.authentication.k8s.io/v1alpha1", "spec": {}, "status": {"expirationTimestamp": "2020-06-24T13:31:06Z", "token": "k8s-aws-v1.aHR0cHM6Ly9zdHMuYW1h…."}}
```

AFAIU the aws-cli version installed in the AMI used by eksctl is too old.

**Comment by thbkrkr** [comment]
And ... eksctl `v0.18.0` [switches to aws eks get-token as the authenticator for Amazon Linux 2](https://github.com/weaveworks/eksctl/blob/ecc5a629aba21b63fd43aaf6f083faf28eabc7cb/docs/release_notes/0.18.0.md).

**Comment by thbkrkr** [comment]
For a k8s cluster in v1.13 using AmazonLinux2 in the eu-central-1 region, the AMI should be [ami-05bf7918ca04ccd1b](https://github.com/weaveworks/eksctl/blob/a7fc5f6e8adc9e5581d7d8d128c7ed806675bb0f/pkg/ami/static_resolver_ami.go#L37).

My nodegroup picked another AMI.
```
> eksctl get nodegroup --region=eu-central-1 --cluster=thb-debug-2 -o json | jq '.[].ImageID' -r
ami-07034b303e1ffc843
```

**Comment by thbkrkr** [comment]
I think there is an issue with the AMI statically resolved by eksctl.

```2020-06-24T17:28:07Z [▶]  resolving AMI using StaticGPUResolver for region eu-central-1, instanceType c5d.2xlarge and imageFamily AmazonLinux2
2020-06-24T17:28:07Z [▶]  can't resolve AMI using StaticGPUResolver as instance type c5d.2xlarge is non-GPU
2020-06-24T17:28:07Z [▶]  resolving AMI using StaticDefaultResolver for region eu-central-1, version 1.14, instanceType c5d.2xlarge and imageFamily AmazonLinux2
2020-06-24T17:28:07Z [ℹ]  nodegroup "ng-1" will use "ami-07034b303e1ffc843" [AmazonLinux2/1.14]
```

The StaticGPUResolver is tried because it's [the first in the list of the static resolvers](https://github.com/weaveworks/eksctl/blob/9bcfafe55ece4334a8cee10c6c27749959c1b021/pkg/ami/resolver.go#L46) but since the instance is non-GPU the StaticDefaultResolver is used.

`aws eks get-token` exists since [`aws-cli/1.16.150`](https://github.com/aws/aws-cli/commit/5fd40b20142413dd236456559f0dee19b1037b10).

[`ami-07034b303e1ffc843`](https://github.com/weaveworks/eksctl/blob/0.21.0/pkg/ami/static_resolver_ami.go#L135) still uses `aws-cli/1.16.102` (see ssh above), even in the latest version `0.22.0`.

The AMIs statically resolved in eksctl have not been updated [since December 17 2019](https://github.com/weaveworks/eksctl/blame/0.22.0/pkg/ami/static_resolver_ami.go#L135) but they will be [soon](https://github.com/weaveworks/eksctl/pull/2339/files#diff-b6434363d4f70a51b5486f71e53fa81c).

To fix this now, we can use [`NodeAMI: auto`](https://eksctl.io/usage/custom-ami-support/). In this way, we got the latest AMI `ami-0454d5737c2365550` with `aws-cli/16.300` and this fixes the issue of the EKS cluster creation.

**Comment by thbkrkr** [comment]
The job is again unstable. Between July 15 and August 11, 8 out of 29 (28%) executions of the job failed on the EKS cluster creation:

```
[2020-08-11T08:50:25.017Z] Error: timed out (after 25m0s) waiting for at least 3 nodes to join the cluster and become ready in "ng-1"
```

It is not sure that this is caused by the same reason as before. I'm going to increase locally the log verbosity to try to understand what is going on.

**Comment by thbkrkr** [comment]
I couldn't reproduce the issue. I managed to create a dozen EKS clusters without encountering any error.

I noticed something suspicious. Failed runs appear to be only those triggered when building release candidates.

The runs triggered by the nightly build are all blue.

![2020-08-13-105138_252x794_scrot](https://user-images.githubusercontent.com/582883/90114726-6ed99a00-dd53-11ea-9599-b30dcc985e2b.png)

Let's create an EKS cluster from the 1.2 branch.
